src/lightning_app/utilities/serializable.py

```diff
@@ -0,0 +1,9 @@
+from typing import Any, Dict
+
+from typing_extensions import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class Hashable(Protocol):
+    def to_dict(self) -> Dict[str, Any]:
+        ...
```

Comment:
in the other libraries we organize these under utilities/types.py. We should probably move this there